geotrellis-sentinelhub/src/main/scala/org/openeo/geotrellissentinelhub/CatalogApi.scala

```diff
@@ -174,21 +221,24 @@ class DefaultCatalogApi(endpoint: String) extends CatalogApi {
           .valueOr(throw _)
       }
 
-      def getFeatures(limit: Int, nextToken: Option[Int]): Map[String, geotrellis.vector.Feature[Geometry, ZonedDateTime]] = {
+
+      def getFeatures(limit: Int, nextToken: Option[Int]): Map[String, geotrellis.vector.Feature[Geometry, FeatureData]] = {
         val page = getFeatureCollectionPage(limit, nextToken)
 
+        var features = Map[String, Feature[Geometry, FeatureData]]()
         // it is assumed the returned geometries are in LatLng
-        val features = page.getAllFeatures[Feature[Geometry, Json]]
-          .mapValues(feature =>
-            feature.mapData { properties =>
-              val Some(datetime) = for {
-                properties <- properties.asObject
-                json <- properties("datetime")
-                datetime <- json.asString
-              } yield ZonedDateTime.parse(datetime, ISO_OFFSET_DATE_TIME)
-
-              datetime
-            })
+        page.features.foreach { fJson =>
```

Comment:
Prefer a `map` (or in this case, `mapValues` because you already have the feature ID) in favor of a `foreach` with a side-effect.

Reply:
`page.features` is a simple `List[JSON]`, so some parsing is needed to convert it to a `Map[String, Feature[...]]`
A fix was made in https://github.com/Open-EO/openeo-geotrellis-extensions/pull/134

Reply:
I see what you mean. I was under the assumption that we could still use `page.getAllFeatures`.